utils.py

```diff
@@ -103,6 +171,52 @@ def green(string):
     print(f"\n\033[01;32m{string}\033[0m", flush=True)
 
 
+def prepare_initrd(architecture, rootfs_format='cpio'):
+    """
+    Returns a decompressed initial ramdisk.
+
+    Parameters:
+        architecture (str): Architecture to download image for.
+        rootfs_format (str): Initrd format ('cpio' or 'ext4')
+    """
+    src = Path(BOOT_UTILS, 'images', architecture,
+               f"rootfs.{rootfs_format}.zst")
+    src.parent.mkdir(exist_ok=True, parents=True)
+
+    # First, make sure that the current user is not rate limited by GitHub,
+    # otherwise the next API call will not return valid information.
+    gh_json_rl = get_gh_json('https://api.github.com/rate_limit')
+    limit = gh_json_rl['resources']['core']['limit']
```

Comment:
move the definition of `limit` into the elif branch that is the sole user of this var. No need to calculate it for the happy path.

Reply:
Done: 929044427808a45a5482f808b46972ea7ea170c3